core/input/input_event.cpp

```diff
@@ -603,12 +643,14 @@ void InputEventKey::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("as_text_keycode"), &InputEventKey::as_text_keycode);
 	ClassDB::bind_method(D_METHOD("as_text_physical_keycode"), &InputEventKey::as_text_physical_keycode);
 	ClassDB::bind_method(D_METHOD("as_text_key_label"), &InputEventKey::as_text_key_label);
+	ClassDB::bind_method(D_METHOD("as_text_location"), &InputEventKey::as_text_location);
 
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "pressed"), "set_pressed", "is_pressed");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "keycode"), "set_keycode", "get_keycode");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "physical_keycode"), "set_physical_keycode", "get_physical_keycode");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "key_label"), "set_key_label", "get_key_label");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "unicode"), "set_unicode", "get_unicode");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "location", PROPERTY_HINT_ENUM, "Either,Left,Right"), "set_location", "get_location");
```

Comment:
Any particular reason to represent `UNKNOWN` as `Either` here?

Reply:
Good question :thinking: 

I believe I defined it this way for the purposes of the inspector, when eg. setting shortcuts on controls:

![Screenshot_20240122_221520](https://github.com/godotengine/godot/assets/441188/772617d7-9929-4b1d-bd9d-973fe90af2ed)

I think the meaning is more clear than were it "unknown", since "unknown" could be interpreted to mean neither left nor right! :sweat_smile: 

I'm not sure if there's anywhere else that this value is used though, where "either" would be less suitable than "unknown"? :thinking:

Reply:
For ones that don't have left/right there's no "either", it'd be "N/A" or "unknown", I'd say "unknown" is better than "either" as it's applicable to left/right cases *and* ones where it doesn't apply, whereas the opposite isn't true

Reply:
I've switched this to "Unknown" :+1: 

FTR, there is still one other "either" in the PR, in the input mapping UI.  Though in that case it only appears for the relevant keys, so should always be appropriate?

![Screenshot_20230805_130332-1](https://github.com/godotengine/godot/assets/441188/17b6cb00-c344-4256-90af-442cf17c1817)

Reply:
Yeah, in this case it seems appropriate. Thanks!

Reply:
I'm a bit late, but how about "Any" instead of "Unknown" or "Either"? Or "Undetermined", but I think "Any" might be the least confusing, and applies for both single or multi location keys.

Also worth flagging, this might be confusing for keys which are duplicated on the numpad, like `*` or `/`. But that's another story :)

Reply:
> I'm a bit late, but how about "Any" instead of "Unknown" or "Either"? Or "Undetermined", but I think "Any" might be the least confusing, and applies for both single or multi location keys.

I actually like that better than "either" too, even though the location selector is only shown for the four relevant modifier keys. "Any" seems so obvious that I don't know why I didn't consider it at the time! :sweat_smile:

Reply:
Then I'd suggest changing it all to `"Any"` / `KEY_LOCATION_ANY`. Then we should be good to merge IMO.

Reply:
Oh hmm, I didn't realise you meant to change the constant name as well! :see_no_evil:  :thinking: 

I'm hesitant to rename the constant too, because the core of this functionality is that we're reporting the location of a specific keypress, and "any" isn't really a location.
"What is the key's location? Any" vs "What is the key's location? Unknown" :confused: 

The constant is only used to mean "any" when doing comparisons in the `InputEvent.is_match` function.

Reply:
Well, what is the location used for aside from match checks? For me "any" means that the InputEventKey with scancode `n` may come from any physical keys that produce this scancode. So both "unknown" and "any" are accurate descriptions of this state, but I'm not sure "unknown" adds anything - as an API user, I would wonder "how do I resolve this uncertainty"? While "any" tells me that it doesn't matter, and not that the information couldn't be found.

Reply:
The perspective I was coming at this API was that of a GDScript user (particularly, me! :sweat_smile: ), where for example in this PR's test project I did:

```gdscript
func _unhandled_input(input: InputEvent):
	if event is InputEventKey and event.location == KEY_LOCATION_LEFT:
		match event.location:
			KEY_LOCATION_LEFT:
				%Left.flip_v = event.is_pressed()
			KEY_LOCATION_RIGHT:
				%Right.flip_v = event.is_pressed()
			KEY_LOCATION_UNKNOWN:
				%Unknown.flip_v = event.is_pressed()
```

I think `KEY_LOCATION_ANY` would make sense if the location were a bitmask, or if we only had comparison functions to access the location (eg. `event.is_location(KEY_LOCATION_ANY)`, but in the context of a property of a `KeyEvent` object I think "any" would be confusing. :confused: 

My assumption was that the user was much more likely to directly encounter the enum under this circumstance, than they would to compare two KeyInput objects explicitly. I would expect such comparisons to be handled internally by the engine after the user just configures the input map and shortcuts through the UI.
This was my reasoning for labelling the value "Either" where this is exposed in the UI, but naming the constant "unknown".

Reply:
How would you feel about "Unspecified"?

Reply:
> How would you feel about "Unspecified"?

The only negative I can think of is that it's a bit longer than "unknown"! :sweat_smile: 

There's also "undefined" as another alternative, but right now I think "unspecified" conveys the meaning as well as or better in the various contexts.

So unless I hear otherwise, I'll do the switch to "unspecified" in my next waking cycle. :+1: 
Though keeping "Any" in the input mapping interface, right?

Reply:
I like "Unspecified" :)

>  Though keeping "Any" in the input mapping interface, right?

Your call, I think both would work fine.

Sorry for repainting this bike shed over and over again :P

Reply:
In the mapping UI, which we only see for twinned keys, I agree that either option would work. We can have contextual text in there. So up to you, I agree.

Reply:
Okay! All `UNKNOWN` locations have been replaced by `UNSPECIFIED` ones :+1: 

> Sorry for repainting this bike shed over and over again :P

You'll get no bikeshaming from me! Naming things is difficult precisely because it's important! :smile: